iroh-metrics/src/core.rs

```diff
@@ -119,17 +141,29 @@ pub trait MObserver {
 /// Internal wrapper to record metrics only if the core is enabled.
 ///
 /// Recording is for single-value metrics, each recorded metric represents a metric value.
-pub(crate) fn record<M>(c: Collector, m: M, v: u64)
+pub fn record<M>(c: &str, m: M, v: u64)
 where
     M: MetricType + std::fmt::Display,
 {
     if CORE.is_enabled() {
-        match c {
-            Collector::Iroh => CORE.iroh_metrics().record(m, v),
-            Collector::Magicsock => CORE.magicsock_metrics().record(m, v),
-            Collector::Netcheck => CORE.netcheck_metrics().record(m, v),
-            Collector::Portmap => CORE.portmap_metrics().record(m, v),
-        };
+        let cc = c.to_string();
+        let mn = m.name().to_string();
+        tokio::task::spawn(async move {
```

Comment:
why spawn?

Reply:
Not an async func, but need to await stuff. Making the above async too makes things a lot worse/incompatible with `once_cell::Lazy`

Reply:
but then you should use https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#method.block_on